src/bookshelf/src/bksw_io_names.erl

```diff
@@ -17,6 +17,8 @@
 
 -module(bksw_io_names).
 
+-compile([nowarn_deprecated_function]).
```

Comment:
It would be better If we update the deprecated fucntions than nowarn flag.

Reply:
@sreepuramsudheer @jashaik We could remove the change, but then the change would come again when we upgrade to erlang 24, which we want to do as soon as possible.

The bigger context is, this issue is fixed in the sigv4 changes.  So this is a bandaid until sigv4 lands.  That was my idea, anyway.  That's why I did nowarn instead of investing time to fix it.

Reply:
> Would it be possible to remove this change as we are still on erlang 22.

I would be happy to do this, but if I understood @PrajaktaPurohit correctly, the purpose of this PR was to move as many commits from the Erlang 24x upgrade as possible (if the commits don't break anything, pipelines remain green, etc) onto main.  You didn't have this context, so I'm just providing it for you :)

Reply:
> It would be better If we update the deprecated fucntions than nowarn flag.

Fixed in upcoming sigv4 changes, if that helps your thoughts on this.

Reply:
We are not updating the Erlang in this PR So please remove this line.This can be part of your Erlang upgrade PR

Reply:
I understand why you are asking for this.  However the full context of this PR is, after extensive consultation with @PrajaktaPurohit, this is a "debug PR" to move as many commits as possible from the "update Erlang" PR into this PR (as long as builds and pipelines remain green) so there are fewer commits in that PR to debug, and so we more easily know where the issues are in that remaining PR.  These are the commits we know are green and don't have any issues.

It's not your fault that you don't have all the context - I made this PR quickly and there was no write-up or context.  I will add it now, and perhaps it will help.

Let me know your thoughts.

Reply:
@jashaik Maybe it is possible that we merge both PRs (this one, and the "upgrade Erlang" PR) at the same time, after both are approved.  If we do that, all the changes land at once.

Reply:
@PrajaktaPurohit Thanks, and no worries.  I'll have a look at the `nowarn`s to see how much work it would be to duplicate the work.